pkg/authorization/static/static.go

```diff
@@ -48,8 +49,15 @@ type staticAuthorizer struct {
 
 // NewStaticAuthorizer creates an authorizer for static SubjectAccessReviews
 func NewStaticAuthorizer(config []StaticAuthorizationConfig) (*staticAuthorizer, error) {
-	for _, c := range config {
-		if c.ResourceRequest != (c.Path == "") {
+	for c := range config {
+		if config[c].User.Groups != nil {
+			groupSet := make(map[string]struct{})
+			for _, group := range config[c].User.Groups {
+				groupSet[group] = struct{}{}
+			}
+			config[c].User.GroupSet = groupSet
+		}
```

Comment:
Use [k8s.io/utils/set](https://github.com/kubernetes/utils/blob/24370beab75816ab1d2b33fba9e288ee667530dd/set/set.go#L28-L35):

```suggestion
		if config[i].User.Groups != nil {
			config[i].User.GroupSet = set.New(config[i].User.Groups...)
		}

```